.github/workflows/dogfood.yaml

```
@@ -72,17 +77,23 @@ jobs:
           tags: "codercom/oss-dogfood:${{ steps.docker-tag-name.outputs.tag }},codercom/oss-dogfood:latest"
 
       - name: Build and push Nix image
-        uses: depot/build-push-action@636daae76684e38c301daa0c5eca1c095b24e780 # v1.14.0
-        with:
-          project: b4q6ltmpzh
-          token: ${{ secrets.DEPOT_TOKEN }}
-          buildx-fallback: true
-          context: "."
-          file: "dogfood/contents/Dockerfile.nix"
-          pull: true
-          save: true
-          push: ${{ github.ref == 'refs/heads/main' }}
-          tags: "codercom/oss-dogfood-nix:${{ steps.docker-tag-name.outputs.tag }},codercom/oss-dogfood-nix:latest"
+        run: |
+          nix build .#dev_image
+
+          # If github.ref is not refs/head/main then exit here early
+           if [ "${{ github.ref }}" != "refs/heads/main" ]; then
+             exit 0
+           fi
```

```suggestion
        if: github.ref == 'refs/heads/main'
        run: |
          nix build .#dev_image

```
May we only run on `main` branch?

---

We always build the images and only push them on the main branch.
This aligns with the 'Build and push Non-Nix image' step, which also builds the image without pushing it and with the behavior before the refactoring.

---

Makes sense.

---

Oh yeah, it would look cleaner/more GitHub Actions native. ✨ 
Let me change that quickly